lib/known_bugs.pm

```diff
@@ -0,0 +1,46 @@
+#$testapi::distri->set_expected_serial_failures(
+#    soft=>{
+#        "AWESOME SOFT MSG 1"=>[qr/gcc version/], 
+#        "AWESOME SOFT MSG 2"=>[qr/insmod error: 1/]
+#    }, 
+#    hard=>{"AWESOME HARD MSG"=>[qr/No iBFT detected/]
+#    }
+#);
+
+use base Exporter;
+use Exporter;
+use strict;
+use testapi;
+
+our @EXPORT = qw(
+    set_serial_failures
+);
+
+sub set_serial_failures {
+    
+    my $serial_failures = [];
+
+    # To add a known bug simply copy and adapt the following line:
+    # push @$serial_failures, {type => soft/hard, message => 'Errormsg', pattern => quotemeta 'ErrorPattern' }
+    # Detect bsc#1093797 on aarch64
+    
+    if (is_sle('=12-SP4') && check_var('ARCH', 'aarch64')) {
+        push @$serial_failures, {type => 'hard', message => 'bsc#1093797', pattern => quotemeta 'Internal error: Oops: 96000006'};
+    }
+
+    if (is_kernel_test()) {
+        my $type = is_ltp_test() ? 'soft' : 'hard';
+        push @$serial_failures, {type => $type, message => 'Kernel Ooops found',             pattern => quotemeta 'Oops:'};
+        push @$serial_failures, {type => $type, message => 'Kernel BUG found',               pattern => qr/kernel BUG at/i};
+        push @$serial_failures, {type => $type, message => 'WARNING CPU in kernel messages', pattern => quotemeta 'WARNING: CPU'};
+        push @$serial_failures, {type => $type, message => 'Kernel stack is corrupted',      pattern => quotemeta 'stack-protector: Kernel stack is corrupted'};
+        push @$serial_failures, {type => $type, message => 'Kernel BUG found',               pattern => quotemeta 'BUG: failure at'};
+        push @$serial_failures, {type => $type, message => 'Kernel Ooops found',             pattern => quotemeta '-[ cut here ]-'};
+    }
+
+    push @$serial_failures, {type => 'soft', message => 'welcome screen found', pattern => quotemeta 'Welcome to SUSE'};
+
+    $testapi::distri->set_expected_serial_failures($serial_failures);
```

Comment:
@DrMullings just for debugging, could you add diag messages before and after set_expected_serial_failures is called?.

Also, serial failures are processed after a test module has been executed... could it be a bug in the implementation?... try adding a shutdown module to your test scenario, and make the failure for the time being, a hard one, so that the test dies. But I guess we'll have to hack a bit [basetest.pm](https://github.com/os-autoinst/os-autoinst/blob/master/basetest.pm#L693)